File: packages/api/start_with_env.js
#!/usr/bin/env node

/**
 * Start API server with environment variables
 */

// Load environment variables first
require('dotenv/config');

const PORT = process.env.PORT || 8080;
const NODE_ENV = process.env.NODE_ENV || 'development';
const ML_BASE_URL = process.env.ML_BASE_URL || 'http://localhost:8000';

console.log('🚀 Starting Agri-Connect API Server...');
console.log(`📍 Port: ${PORT}`);
console.log(`🌍 Environment: ${NODE_ENV}`);
console.log(`🤖 ML Service: ${ML_BASE_URL}`);
console.log(`🔗 API URL: http://localhost:${PORT}`);
console.log(`📚 Health Check: http://localhost:${PORT}/health`);
console.log('-'.repeat(50));

// Start the server
require('./src/index.ts');

File: packages/ml/start_with_env.py
#!/usr/bin/env python3
"""
Start ML service with environment variables
"""
import os
import sys
from dotenv import load_dotenv

# Load environment variables
load_dotenv()

# Get configuration from environment
ML_SERVICE_HOST = os.environ.get('ML_SERVICE_HOST', '0.0.0.0')
ML_SERVICE_PORT = int(os.environ.get('ML_SERVICE_PORT', '8000'))

if __name__ == "__main__":
    import uvicorn
    from main import app
    
    print(f"🚀 Starting ML Service...")
    print(f"📍 Host: {ML_SERVICE_HOST}")
    print(f"📍 Port: {ML_SERVICE_PORT}")
    print(f"🔗 URL: http://{ML_SERVICE_HOST}:{ML_SERVICE_PORT}")
    print(f"📚 Docs: http://{ML_SERVICE_HOST}:{ML_SERVICE_PORT}/docs")
    print("-" * 50)
    
    uvicorn.run(
        app,
        host=ML_SERVICE_HOST,
        port=ML_SERVICE_PORT,
        reload=True,
        log_level="info"
    )
